rdt/transformers/categorical.py

```diff
@@ -425,8 +437,41 @@ class LabelEncoder(BaseTransformer):
     values_to_categories = None
     categories_to_values = None
 
-    def __init__(self, add_noise=False):
+    def __init__(self, add_noise=False, order_by=None):
         self.add_noise = add_noise
+        if order_by not in [None, 'alphabetical', 'numerical_value']:
+            raise Error(
+                "order_by must be one of the following values: None, 'numerical_value' or "
+                "'alphabetical'"
+            )
+
+        self.order_by = order_by
+
+    def _order_categories(self, unique_data):
```

Comment:
With the new logic, I think you could reorganize the method like this:

```python3
    def _order_categories(self, unique_data):
        if self.order_by == 'alphabetical':
            if unique_data.dtype.type not in [np.str_, np.object_]:
                raise Error("The data must be of type string if order_by is 'alphabetical'.")

        elif self.order_by == 'numerical_value':
            if not np.issubdtype(unique_data.dtype.type, np.number):
                raise Error(
                    'The data must be numerical or able to be casted as a float if order_by '
                    "is 'numerical_value'."
                )

        if self.order_by is not None:
            nans = pd.isna(unique_data)
            nan_rep = unique_data[nans]
            unique_data = np.sort(unique_data[~nans])
            unique_data = np.append(unique_data, nan_rep)

        return unique_data
```
So validate the data first, then do the sorting.

Reply:
The reason I only do this in the 'alphabetical' section is that `np.sort` automatically sorts nans last with floats. It just doesn't work when it has to compare strings to nan, so in the string case I removed the nans and add it back in after the sorting

Reply:
Sure, I just think the code could maybe be better organized. So instead of applying only to alphabetical, I applied to both to make the code cleaner. I don't think we lose much, if any, speed by doing it this way (since the only difference is that we manually put the nans at the end instead of letting np handle it). Removing the `if nans.any():` is the same thing, I don't think it gains any speed and it just makes the code slightly more complicated, which is why I think it could be removed.

Either way works fine though, the performance and results should be the same.